src/Microsoft.Identity.Web.Certificate/DefaultCredentialsLoader.cs

```diff
@@ -15,6 +16,9 @@ namespace Microsoft.Identity.Web
     public class DefaultCredentialsLoader : ICredentialsLoader
     {
         ILogger<DefaultCredentialsLoader>? _logger;
+        
+        private readonly Dictionary<string, SemaphoreSlim> _loadingSemaphores = new Dictionary<string, SemaphoreSlim>();
```

Comment:
How does the Dictionary<string, SemaphoreSlim> help us?

Reply:
A semaphore is created per CredentialDescription. A thread can safely acquire a semaphore for a specific CredentialDescription. Other threads are free to acquire a semaphore to load the certificate for other CredentialDescriptions. Aim here is to minimize thread blocking.